crates/rustc_codegen_spirv/src/codegen_cx/entry.rs

```diff
@@ -663,6 +665,43 @@ impl<'tcx> CodegenCx<'tcx> {
             }
         }
 
+        let is_subpass_input = match self.lookup_type(value_spirv_type) {
+            SpirvType::Image {
+                dim: Dim::DimSubpassData,
+                ..
+            } => true,
+            SpirvType::RuntimeArray { element: elt, .. }
+            | SpirvType::Array { element: elt, .. } => matches!(
+                self.lookup_type(elt),
+                SpirvType::Image {
+                    dim: Dim::DimSubpassData,
+                    ..
+                }
+            ),
+            _ => false,
+        };
+        if let Some(attachment_index) = attrs.input_attachment_index.map(|attr| attr.value) {
+            if is_subpass_input {
+                self.emit_global().capability(Capability::InputAttachment);
```

Comment:
Ah, with https://github.com/EmbarkStudios/rust-gpu/pull/630, we no longer automatically add capabilities for users, as it's a really bad user experience. Instead, check `self.builder.has_capability(Capability::InputAttachment)`, and emit an error if false.